lib/phoenix_live_view/channel.ex

```diff
@@ -508,6 +510,19 @@ defmodule Phoenix.LiveView.Channel do
     {:stop, :shutdown, :no_session}
   end
 
+  defp verify_flash(endpoint, verified, params) do
+    flash_token = params["flash"]
+    verified_flash = verified[:flash]
+
+    # verified_flash is fetched from the disconnected render.
+    # params["flash"] is sent on live redirects and therefore has higher priority.
+    cond do
+      flash_token -> Phoenix.LiveView.Flash.verify(endpoint, flash_token)
```

Comment:
So this may return nil, which will set the flash to nil. Ideally you would do `|| %{}`, but there is another place where you call `verify || %{}`, so I would suggest for verify to simply return `%{}` instead of `nil` and simplify the callers.